lib/manifest.go

```diff
@@ -0,0 +1,178 @@
+package lib
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Manifest stores Kubernetes object in Raw from a file.
+// It stores the GroupVersionKind for the manifest.
+type Manifest struct {
+	Raw []byte
+	GVK schema.GroupVersionKind
+
+	obj *unstructured.Unstructured
+}
+
+// UnmarshalJSON unmarshals bytea of single kubernetes object to Manifest.
+func (m *Manifest) UnmarshalJSON(in []byte) error {
+	if m == nil {
+		return errors.New("Manifest: UnmarshalJSON on nil pointer")
+	}
+	if !bytes.Equal(in, []byte("null")) {
```

Comment:
nit: why need to special case this? won't this be handled by the Decode()?

Reply:
update it add comment. look [here](https://github.com/openshift/cluster-version-operator/pull/10/commits/109df8379ae64172ca9ecf4ce2061f71b2a8009c#diff-d1961d4f655d836d50969836a0f51785R96)